sensor/common/compliance/node_inventory_handler_impl.go

```diff
@@ -283,23 +298,141 @@ func (c *nodeInventoryHandlerImpl) sendNodeIndex(toC chan<- *message.ExpiringMes
 		log.Debugf("Empty IndexReport - not sending to Central")
 		return
 	}
+
+	isRHCOS, version, err := c.nodeRHCOSMatcher.GetRHCOSVersion(indexWrap.NodeName)
+	if err != nil {
+		log.Warnf("Unable to determine RHCOS version for node %q: %v", indexWrap.NodeName, err)
+		isRHCOS = false
+	}
+	log.Debugf("Node=%q discovered RHCOS=%t rhcos-version=%q", indexWrap.NodeName, isRHCOS, version)
+
 	select {
 	case <-c.stopper.Flow().StopRequested():
-	case toC <- message.New(&central.MsgFromSensor{
-		Msg: &central.MsgFromSensor_Event{
-			Event: &central.SensorEvent{
-				Id: indexWrap.NodeID,
-				// ResourceAction_UNSET_ACTION_RESOURCE is the only one supported by Central 4.6 and older.
-				// This can be changed to CREATE or UPDATE for Sensor 4.8 or when Central 4.6 is out of support.
-				Action: central.ResourceAction_UNSET_ACTION_RESOURCE,
-				Resource: &central.SensorEvent_IndexReport{
-					IndexReport: indexWrap.IndexReport,
+	default:
+		defer func() {
+			log.Debugf("Sent IndexReport to Central")
+			metrics.ObserveReceivedNodeIndex(indexWrap.NodeName) // keeping for compatibility with 4.6. Remove in 4.8
+			metrics.ObserveNodeScan(indexWrap.NodeName, metrics.NodeScanTypeNodeIndex, metrics.NodeScanOperationSendToCentral)
+		}()
+		irWrapperFunc := passthruRPM
+		if isRHCOS {
+			log.Debugf("Attaching OCI entry for 'rhcos' to index-report: version=%s", version)
+			irWrapperFunc = attachRPMtoRHCOS
+		}
+		toC <- message.New(&central.MsgFromSensor{
+			Msg: &central.MsgFromSensor_Event{
+				Event: &central.SensorEvent{
+					Id: indexWrap.NodeID,
+					// ResourceAction_UNSET_ACTION_RESOURCE is the only one supported by Central 4.6 and older.
+					// This can be changed to CREATE or UPDATE for Sensor 4.8 or when Central 4.6 is out of support.
+					Action: central.ResourceAction_UNSET_ACTION_RESOURCE,
+					Resource: &central.SensorEvent_IndexReport{
+						IndexReport: irWrapperFunc(version, indexWrap.IndexReport),
+					},
+				},
+			},
+		})
+	}
+}
+
+func normalizeVersion(version string) []int32 {
+	fields := strings.Split(version, ".")
+	switch len(fields) {
+	case 0:
+		return []int32{0, 0, 0}
+	case 1:
+		i, err := strconv.ParseInt(fields[0], 10, 32)
+		if err == nil {
+			return []int32{int32(i), 0, 0}
+		}
+	}
+	i1, err1 := strconv.ParseInt(fields[0], 10, 32)
+	if err1 != nil {
+		i1 = 0
+	}
+	i2, err2 := strconv.ParseInt(fields[1], 10, 32)
+	if err2 != nil {
+		i2 = 0
+	}
+	// Only two first fields matter for the initial db query.
+	// The results of that query will be filtered further using string compare of the Version field.
+	return []int32{int32(i1), int32(i2), 0}
+}
+
+func passthruRPM(_ string, rpm *v4.IndexReport) *v4.IndexReport {
```

Comment:
I pitch to rename this function. Going in, I expected it to do something with RPMs of the report, but it doesn't. Maybe something like
- `noop`
- `noupdate`
- `reportPassthrough`
?

Seeing that the `attachRPMtoRHCOS` does technically do some RPM work, I can see the name a bit better there, but I feel there is something out there that is better fitted. 
We're not really attaching an RPM to RHCOS, are we? We're replacing the incoming IndexReport with one that only contains an RHCOS component, right?

Reply:
In `attachRPMtoRHCOS`, we actually copy the contents of the RPM index-report into the index-report with single rhcos package. I think now, that we could do it the other way round, but it does not matter much.

I had a name with `noop` earlier, maybe I will craft something with that name.

Reply:
Addressed in [d339131](https://github.com/stackrox/stackrox/pull/13686/commits/d3391318b504b10c13b7b5da0a5de4649301b2d7)